### PR TITLE
Machinegod Cult [WIP][DO NOT MERGE]

### PR DESCRIPTION
**Pull request by VelardAmakar**
aye i'm putting stuff here to display progress and to have it all laid out when i finish
you can watch my progress [here](https://trello.com/c/v7cmdXSI/28-machinegod-cult)
get more info [here](https://docs.google.com/document/d/1Lkp-7MyzX3_BG1Gi7pDMJMZNE5SJTI0cCKlBIyrRXEI/edit#heading=h.5or1u1myu0h9) and [here](https://docs.google.com/spreadsheets/d/1yReHQohhT-a5QEdyfYARavV-UbuP3A3yadL11bNLFbo/edit#gid=127733079)
and when i have a forum thread up for this it'll be [here]()

i'm coding it, @RemieRichards is helping my dumb ass
@SkowronX and @FuryMcFlurry are doing sprites
revenant

### CODE REVIEW

**Comment by Exxion** [comment]
There's a distinct possibility that those Wraith's Spectacles won't play well with harm-labeling because I probably didn't do too great a job on that system
You can probably do nothing and nobody will notice because I may have not exactly added harm-labeling too the changelog, so it's unlikely anyone who doesn't look at Github even knows about the feature.
Or you could just disable harm-labeling of them by setting the min_harm_label variable to 0.

**Comment by VelardAmakar** [comment]
explain harm-labeling to me

**Comment by Exxion** [comment]
Certain items (at the moment, flashers, flashes, film cameras, guns, and most types of glasses) can be disabled by using a labeler on them while on harm intent. They are re-enabled by removing the label, of course. The idea is that you cover some important part of the item with a label. (The bulb on the flashers/flashes, the lens on the film cameras, the trigger on the guns, and the lenses on the glasses.) It only works if the label is at least a certain length (min_harm_label) but it is disabled for an item if its min_harm_label is 0. It defaults to 0. However, these, being a subtype of obj/item/glasses, will inherit min_harm_label=12 from obj/item/glasses. They will also inherit the generic glasses harm-label behavior, which is just add BLIND to vision_flags (of the glasses, not the wearer) when harm-labeled and remove it when the offending label is removed.
If you do want to actually have harm-labeling work, you can change the harm_label_update proc (called when the item is harm-labeled or the label is removed) and/or change the behavior of the item if(min_harm_label && harm_labeled >= min_harm_label).
For bonus points, you can also alter harm_label_examine if you want, but I think the default for glasses would make about as much sense in this case as anything else. If you do want to change it from its default, it is a list containing two strings. The first is the message to be displayed on examine if someone tried to harm-label it but the label was too short, and the second is the message to be displayed on examine if successfully harm-labeled.

Might be worth noting that mesons and thermals (except in monocle form) currently use the default glasses harm-label behavior, which means they retain their special effects, but the user loses all other sight.
If I understand and remember correctly, these Wraith's Spectacles would, in their current state, be unaffected by harm-labeling other than gaining some text in their examine messages.

**Comment by RemieRichards** [comment]
Holy fucking shit this is cool.

Real nice work on the lore :+1:

**Comment by VelardAmakar** [comment]
@RemieRichards aaaa thank you
feel free to give feedback or questions or anything

@Exxion well shit
i might see about that sometime
that seems really weird but useful

**Comment by RemieRichards** [comment]
I know @SkowronX is doing the sprites (and I'm sure they can do a better job) but I was too hype
Stuff

**Comment by VelardAmakar** [comment]
that's cool as hell
i'm not a good spriter either, but skowron is hella good at it
if you wanna talk more on the project i'm in #vgstation some of the time and TETRASWAGMATON on steam most of the time

your assistance with coding in some areas would be amazing, whatever you can do to assist would be just awesome

**Comment by RemieRichards** [comment]
Oh hey I have a pending steam invite from you, sorry about not seeing that, haha

File: code/game/gamemodes/machinegod/machinegod.dm [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+
+/datum/game_mode/machinegod
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod.dm]: Would it not be easier to inherit from cult?
You get a lot of the setup code, surely, and I notice a few copied things (harvest_target).

**Comment by VelardAmakar** [comment]
i barely know shit about gamemode code
i will need to learn a bit more before i can seriously attempt to work on the appropriate datums

**Comment by VelardAmakar** [comment]
UPDATED
2/36 POWERS COMPLETE

**Comment by RemieRichards** [comment]
Progress!

**Comment by VelardAmakar** [comment]
updated

File: code/game/gamemodes/machinegod/sigils.dm [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		M << "<span class='warning'>The null rod negates the sigil's power.</span>"
+		return
+
+	var/turf/T = get_turf(M)
+	T.turf_animation('icons/obj/clockwork/sigils.dmi',"pulse",0, 0, 5, 'sound/machines/notify.ogg', actcolor)
+	color = actcolor
+	spawn(10)
+		color = initial(color)
+
+/obj/effect/sigil/proc/findNullRod(var/atom/target)
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/sigils.dm]: Isn't this copied?

Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: it is.

File: code/game/gamemodes/machinegod/sigils.dm [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		else
+			activation(M)*/
+
+/obj/effect/sigil/cultify() //PURGE
+	qdel(src)
+
+/obj/effect/sigil/proc/activation(var/mob/M as mob) //What does it do when it's triggered?
+	if(M.stat==DEAD)
+		return
+	nullblock = 0
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: make this a local var ```var/nullblock = 0```

File: code/game/gamemodes/machinegod/sigils.dm [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	for(var/turf/TR in range(src,1))
+		findNullRod(TR)
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: change this to 
```
if(findNullRod(TR))
   nullblock = 1
   break
```

File: code/game/gamemodes/machinegod/sigils.dm [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+	var/atom/movable/overlay/h_animation = null
+	var/nullblock = 0
+	var/culttrigger = 0 //Can Ratvar cultists trigger this?
+
+/obj/effect/sigil/Crossed(AM as mob|obj)
+	Bumped(AM)
+
+/obj/effect/sigil/Bumped(mob/M as mob|obj)
+	if(istype(M, /mob/living/carbon/human) || istype(M, /mob/living/carbon/monkey) || istype(M, /mob/living/silicon))
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: replace with ```if(iscarbon(M) || issilicon(M))```

File: code/game/gamemodes/machinegod/sigils.dm [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+			if(culttrigger)
+				activation(M)
+		else
+			activation(M)*/
+
+/obj/effect/sigil/cultify() //PURGE
+	qdel(src)
+
+/obj/effect/sigil/proc/activation(var/mob/M as mob) //What does it do when it's triggered?
+	if(M.stat==DEAD)
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: Should probably be just ```if(M.stat)``` since I dont think you want unconscious people to use it.

Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/sigils.dm]: I do, in one circumstance.

File: code/game/gamemodes/machinegod/sigils.dm (hunk 2 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/sigils.dm]: correct

**Comment by VelardAmakar** [comment]
added the spear
progress

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 7]
```diff
@@ -0,0 +1,372 @@
...
+			H << 'sound/items/healthanalyzer.ogg'
+			H << "The lens lights up."
+			eyeprot = -1
+			if(H.client)
+				H.client.mouse_pointer_icon = file("icons/effects/visor_reticule.dmi")
+		H.update_inv_glasses()
+
+/obj/item/clothing/glasses/judicialvisor/ranged_weapon(var/atom/A, mob/living/carbon/human/wearer as mob)
+	if(cooldown)
+		wearer << "<span class='sinister'>\"Have patience. It's not ready yet.\"</span>"
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: You should consider making these a separate span in gold or something.

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 8]
```diff
@@ -0,0 +1,372 @@
...
+			user.UpdateDamageIcon()*/
+
+/obj/item/weapon/spear/clockspear/pickup(mob/living/user as mob)
+	/*if(!isclockcult(user)) NOT IMPLEMENTED YET -velardamakar
+		user << "<span class='danger'>An overwhelming feeling of dread comes over you as you pick up the cultist's spear. It would be wise to be rid of this weapon quickly.</span>"
+		user.Dizzy(120)*/
+	if(iscultist(user))
+		user << "<span class='sinister'>\"Does a savage like you even know how to use that thing?\"</span>"
+		var/organ = ((user.hand ? "l_":"r_") + "hand")
+		var/datum/organ/external/affecting = user:get_organ(organ)
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: :::::::::::::::::

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+	if(!istype(M)) return
+	if(iscultist(M))
+		M.take_organ_damage(15)
+		M.Stun(1)
+	if(issilicon(M))
+		M.take_organ_damage(8)
+		M.Stun(2)
+	qdel(src)
+
+/spell/clockspear
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: Just use the basic framework for conjuring.
It even does temporary items for you.

File: code/game/gamemodes/machinegod/sigils.dm [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+	color = "#B39700"
+	var/actcolor = ""
+	var/visibility = 0
+	unacidable = 1
+	layer = TURF_LAYER
+
+	var/dead=0 // For cascade and whatnot.
+
+	var/atom/movable/overlay/h_animation = null
+	var/nullblock = 0
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: remove this, it's now unused

File: code/game/gamemodes/machinegod/sigils.dm (hunk 2 above)
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/sigils.dm]: No, not correct!
Very wrong!

Comment (RemieRichards) [inline code/game/gamemodes/machinegod/sigils.dm]: With the change I asked you to make you can now remove this copy-pasted proc.

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 7 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: maybe, haven't decided yet

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 8 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: i'll rectify it once i figure out a clean way to do it

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 9 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: i'll have to modify it slightly to suit my needs but you're correct

File: code/game/gamemodes/machinegod/sigils.dm (hunk 10 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/sigils.dm]: will do

**Comment by RemieRichards** [comment]
@VelardAmakar https://github.com/VelardAmakar/vgstation13/pull/6 My PR cleaning up Clockcult and generic gamemode icon code.

**Comment by VelardAmakar** [comment]
remie da bes

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 11]
```diff
@@ -0,0 +1,375 @@
...
+	playsound(user,'sound/effects/evolve.ogg',100,1)
+	for(var/turf/T in get_area(user))
+		for(var/obj/machinery/light/L in T.contents)
+			if(L && prob(20)) L.flicker()
+
+
+//DEBUG ITEM
+/obj/item/weapon/clockcheat
+	name = "ZOOP"
+	desc = "ZOP"
```
Comment (RemieRichards) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: pudding pop!

**Comment by SkowronX** [comment]
May as well post this here too. Tier 4 summon mob sprites coming soon

revenants

**Comment by VelardAmakar** [comment]
UPDATE

**Comment by VelardAmakar** [comment]
>conflicts because git fucked up my merge

**Comment by SkowronX** [comment]
fix it yo

**Comment by VelardAmakar** [comment]
fix'd

File: code/game/gamemodes/machinegod/machinegod_structures.dm [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+	if(isclockcult(L)) return
+
+	var/event_id = pick(2;"hallucination", 1;"clumsiness", 2;"stutter", 1;"vomit")
+	switch(event_id)
+		if("hallucination")
+			L.hallucination = max(L.hallucination, 35)
+		if("clumsiness")
+			if(!(M_CLUMSY in L.mutations))
+				L.mutations.Add(M_CLUMSY)
+		if("disorient")
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_structures.dm]: This isn't consistent with the pick.

File: code/game/gamemodes/machinegod/machinegod_items.dm [hunk 13]
```diff
@@ -0,0 +1,427 @@
+/obj/item/clock_component
```
Comment (ComicIronic) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: As a general comment, combining all this stuff together in one file makes it pretty hard to read.
The spells should go in the spell folder, and then I think you should split this into components and equipment.

File: code/game/gamemodes/machinegod/machinegod_structures.dm (hunk 12 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_structures.dm]: fun fact: i am a huge spaz

File: code/game/gamemodes/machinegod/machinegod_items.dm (hunk 13 above)
Comment (VelardAmakar) [inline code/game/gamemodes/machinegod/machinegod_items.dm]: i'll separate this shit further down the line, right now it's smashed together for my convenience more than anything

**Comment by SkowronX** [comment]
Ngaaaahhh

**Comment by FuryMcFlurry** [comment]
??

**Comment by clusterfack** [comment]
Handle ranged attack would be a nice addition to be an atomic pull separate from this.

**Comment by VelardAmakar** [comment]
Added the gateway finally

**Comment by VelardAmakar** [comment]
fucking git

**Comment by VelardAmakar** [comment]
how'd i close it tho

**Comment by VelardAmakar** [comment]
oh i get it

**Comment by VelardAmakar** [comment]
finally fixed it

**Comment by SkowronX** [comment]
Why close it?

**Comment by RemieRichards** [comment]
@SkowronX from my email notifications it looks like he did it on accident.

**Comment by clusterfack** [comment]
He probably deleted the branch by accident

**Comment by VelardAmakar** [comment]
shit was automerged
it won't open again until i push another commit

**Comment by VelardAmakar** [comment]
ty github
guess i'll just use the new one

**Comment by clusterfack** [comment]
0/0 lines changed, hahaha

**Pull request closed without merging**